app/controllers/events_controller.rb

```diff
@@ -0,0 +1,112 @@
+class EventsController < ApplicationController
+  before_action :set_event, only: %i[show edit update destroy]
+  before_action :set_current_user, only: %i[show index new attend_events show_events]
+
+  # GET /events
+  # GET /events.json
+  def index
+    @events = Event.all
+    @upcoming_events = Event.upcoming_events
+    @previous_events = Event.previous_events
+  end
+
+  # GET /events/1
+  # GET /events/1.json
+  def show
+    @creator = creator
+  end
+
+  # GET /events/new
+  def new
+    @event = Event.new
+  end
+
+  # GET /events/1/edit
+  def edit; end
+
+  # POST /events
+  # POST /events.json
+  def create
+    @event = current_user.events_created.build(event_params)
+
+    respond_to do |format|
+      if @event.save
+        format.html { redirect_to @event, notice: 'Event was successfully created.' }
+        format.json { render :show, status: :created, location: @event }
+      else
+        format.html { render :new }
+        format.json { render json: @event.errors, status: :unprocessable_entity }
+      end
+    end
```

Comment:
There's no need to render both formats if you are using normal html.erb views
This happens because you created the app using the scaffold option. I advise against that since it creates unnecessary codes and files.